### PR TITLE
feat: File globs using mattn/go-zglob library

### LINKED ISSUE #14: glob patterns for files/config_files
[This](https://github.com/goreleaser/nfpm/blob/0c6b7821812f13134ac062a82b45c863ada261a0/deb/deb.go#L124) had me thinking. Would you consider adding glob patterns to match files/folders? The obvious choice is to use the [path/filepath](https://golang.org/pkg/path/filepath/#Glob) package. However it [doesn't support the ** glob pattern](https://github.com/golang/go/issues/11862) (at least for now). Maybe one could resolve to [this library called zglob](https://github.com/mattn/go-zglob)? Any opinions?

### PR DESCRIPTION
**Pull request by tympanix**
Added file globs to deb and rpm packager. Since the destination for
a glob can not be a single file, the files collected by the glob are
stored in: join(dst, trim(lcp(src), src)) (read: lcp = longest common
prefix). Thereby the longest common prefix is removed from every path in
the globbed files and joined with the destination path. Implementation
is backwards compatible (i.e. non-globs will work as before)

Closes #14

### CODE REVIEW

File: pkgutil/pkgutil.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package pkgutil
```
Comment (caarlos0) [inline pkgutil/pkgutil.go]: this is not a good package name... maybe `glob`?

File: pkgutil/pkgutil.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	"fmt"
+	"os"
+	"path/filepath"
+
+	zglob "github.com/mattn/go-zglob"
+)
+
+// LongestCommonPrefix returns the longest prefix of all strings the argument
+// slice. If the slice is empty the empty string is returned
+func LongestCommonPrefix(strs []string) string {
```
Comment (caarlos0) [inline pkgutil/pkgutil.go]: this function could be private...

File: pkgutil/pkgutil.go (hunk 1 above)
Comment (caarlos0) [reply pkgutil/pkgutil.go]: also missing docs

File: pkgutil/pkgutil.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+func Glob(glob, dst string) (map[string]string, error) {
+	matches, err := zglob.Glob(glob)
+	if err != nil {
+		return nil, err
+	}
+	if len(matches) == 0 {
+		return nil, fmt.Errorf("no files matched by: %s", glob)
+	}
+	files := make(map[string]string)
+	prefix := LongestCommonPrefix(matches)
```
Comment (caarlos0) [inline pkgutil/pkgutil.go]: i don't get why we need this..

Comment (caarlos0) [reply pkgutil/pkgutil.go]: oh, nevermind, just realized it...

Comment (tympanix) [reply pkgutil/pkgutil.go]: to be clear: by removing the longest prefix we enable the user to have the greatest control over the resulting destination path.

**Comment by tympanix** [comment]
Bump. Are we good or waiting for other parts of nfpm to mature first?

**Comment by caarlos0** [comment]
Hey @tympanix, I still need to test this within goreleaser and decide if it's better to put this here or there...

sorry for the delay, crazy shit happening here

**Comment by tympanix** [comment]
No worries. I would personally love to have the feature here because it promotes npfm as a standalone tool as well. Win win. Looking forward to it.

**Comment by caarlos0** [comment]
This pull request has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**